enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/utils/derived-colors-for-embed-flow.ts

```diff
@@ -0,0 +1,87 @@
+import type { MetabaseColors, MetabaseTheme } from "embedding-sdk";
+import type { MetabaseColor } from "metabase/embedding-sdk/theme";
+import { applyColorOperation } from "metabase/embedding-sdk/theme/dynamic-css-vars";
+import { SDK_TO_MAIN_APP_COLORS_MAPPING } from "metabase/embedding-sdk/theme/embedding-color-palette";
+import { colors, isDark } from "metabase/lib/colors";
+import type { ColorName, ColorPalette } from "metabase/lib/colors/types";
+
+import { DYNAMIC_SDK_DEFAULT_COLORS_CONFIG } from "./dynamic-sdk-color-defaults";
+
+/**
+ * Generate derived colors for SDK colors that the user did not define in the embed flow.
+ * This is a breaking-change-free implementation that works with MetabaseTheme format
+ * instead of MantineThemeOverride.
+ *
+ * @param theme The current MetabaseTheme with user-defined colors
+ * @param appColors Optional application colors from instance settings
+ * @returns A new MetabaseTheme with derived colors added
+ */
+export function getDerivedDefaultColorsForEmbedFlow(
+  theme: MetabaseTheme,
+  appColors: ColorPalette = {},
+): MetabaseTheme {
+  const themeColors = theme.colors ?? {};
+
+  const getSourceColor = (colorName: MetabaseColor): string => {
```

Comment:
This is the color we use to sample colors to derive. Usually this is `background`.